### PR TITLE
add libnoise/1.0.0

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libnoise/all/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+endif()
+
+include(GNUInstallDirs)
+install(TARGETS ${PROJECT_NAME}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+
+install(FILES ${NOISE_HDR_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/noise)
```
Comment (SpaceIm) [inline recipes/libnoise/all/CMakeLists.txt:36]: oops, it does not honor include hierarchy

File: recipes/libnoise/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.4)
+project(noise)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(NOISE_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder/noise/src)
+
+file(GLOB NOISE_SRC_FILES
+    ${NOISE_SRC_DIR}/*.cpp
```
Comment (madebr) [inline recipes/libnoise/all/CMakeLists.txt:10]: The archive contains: `src/win32/libnoise.def`.
It contains a lot of symbols though.

So I think using `WINDOWS_EXPORT_ALL_SYMBOLS` is good enough.
As long as there are no static variables not exported.

**Review by madebr — APPROVED** [review]

File: recipes/libnoise/all/CMakeLists.txt (hunk 2 above)
Comment (SpaceIm) [reply recipes/libnoise/all/CMakeLists.txt:10]: Yes you're right. And actually, looking at source code, all symbols are exported anyway. No static variables by the way.

Comment (madebr) [reply recipes/libnoise/all/CMakeLists.txt:10]: So let's keep it like it is.

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**